changelog/2800.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+Fixed a potential data loss scenario when using ``HTTPResponse.read()`` with ``decode_content=True``
+and then subsequently calling the method with ``decode_content=False``.
```

Comment:
@sethmlarson nit: should we replace "Fixed" with "Prevented"? It's slightly more precise but maybe you only want to use a few verbs in the changelog.

Reply:
My convention of only using a few verbs comes from keepachangelog.com. I agree that this wording isn't right: it's a "change" not a fix.

Maybe something like:

```
Changed ``HTTPResponse.read()`` to raise an error when calling with ``decode_content=False`` after using ``decode_content=True`` to prevent data loss.
```

Reply:
Sounds great!

Reply:
And now it looks great:

<img width="739" alt="Capture d’écran 2022-11-20 à 21 43 26" src="https://user-images.githubusercontent.com/42327/202917466-57cd39c7-7fbc-4f2d-b8d4-ec270e1faf3e.png">